kernel/src/main/scala/cats/kernel/std/list.scala

```diff
@@ -0,0 +1,95 @@
+package cats.kernel
+package std
+
+import scala.annotation.tailrec
+import scala.collection.mutable
+
+package object list extends ListInstances
+
+trait ListInstances extends ListInstances1 {
+  implicit def listOrder[A: Order] = new ListOrder[A]
+  implicit def listMonoid[A] = new ListMonoid[A]
+}
+
+trait ListInstances1 extends ListInstances2 {
```

Comment:
I'm inclined to make traits like `ListInstances1` and `ListInstances2` `sealed` and `private[cats.kernel.std]` just to signify that they are an internal detail and not really meant to be used directly. I think we are taking this approach in most of cats.